src/v/kafka/server/materialized_partition.h

```diff
@@ -54,13 +54,14 @@ class materialized_partition final : public kafka::partition_proxy::impl {
     ss::future<storage::translating_reader> make_reader(
       storage::log_reader_config cfg,
       std::optional<model::timeout_clock::time_point>) final {
-        co_return storage::translating_reader(co_await _log.make_reader(cfg));
+        co_return storage::translating_reader(
+          co_await _partition->make_reader(cfg));
     }
 
     ss::future<std::optional<storage::timequery_result>>
     timequery(model::timestamp ts, ss::io_priority_class io_pc) final {
-        storage::timequery_config cfg(ts, _log.offsets().dirty_offset, io_pc);
-        return _log.timequery(cfg);
+        storage::timequery_config cfg(ts, last_stable_offset(), io_pc);
```

Comment:
take a look at some of Michal's recent commits. I think this offset bound here changed.

Reply:
Yup this has been changed in my most recent rebase, was waiting for you to finish review before i pushed.